include/zephyr/sys/util.h

```diff
@@ -44,20 +44,30 @@ extern "C" {
 /** @brief Cast @p x, a signed integer, to a <tt>void*</tt>. */
 #define INT_TO_POINTER(x)  ((void *) (intptr_t) (x))
 
-#if !(defined(__CHAR_BIT__) && defined(__SIZEOF_LONG__))
+#if !(defined(__CHAR_BIT__) && defined(__SIZEOF_LONG__) && defined(__SIZEOF_LONG_LONG__))
```

Comment:
This is going to cause new build errors if someone is building with a toolchain that doesn't define __SIZEOF_LONG_LONG__ (which AFAIK is a gcc-ism, not part of a C standard).  It's probably unlikely, but something to be aware of.

Reply:
These are the other options I see (if sizeof isn't defined)

#define GENMASK64(h, l) \
	(((~0ULL) - (1ULL << (l)) + 1) & (~0ULL >> (64 - 1 - (h))))

#define GENMASK64(h, l) \
    ((h) == 63 && (l) == 0) ? \
        ~0ULL : \
        (((1ULL << ((h)-(l)+1)) - 1) << (l))